src/ddmd/traits.d

```diff
@@ -370,6 +373,73 @@ extern (C++) Expression pointerBitmap(TraitsExp e)
     return ale;
 }
 
+extern (C++) Expression externDmangle(TraitsExp e, Scope* sc)
```

Comment:
This is needed for [mangleFunc](https://dlang.org/library/core/demangle/mangle_func.html) and [mangle](https://dlang.org/library/core/demangle/mangle.html) whose primary use-case is to enforce correct typing when loading symbols from shared libraries at runtime.

Guess we can implement that in core.demangle at some point, as that's somewhat complex it's nice to have it as a trait for now.

This needs to support normal types and function pointers as well, not only mangling of functions.